src/arithmetic/string_funcs/string_funcs.c

```diff
@@ -237,20 +238,51 @@ SIValue AR_JOIN(SIValue *argv, int argc, void *private_data) {
 		str_len += strlen(str.stringval);
 	}
 
-	int cur_len = 0;
-	char *res = rm_malloc(str_len + 1);
+	uint cur_len = 0;
+	char *base = rm_malloc(str_len + 1);
+	char *res = base;
+
 	for(uint i = 0; i < count - 1; i++) {
+		
 		SIValue str = SIArray_Get(list, i);
-		memcpy(res + cur_len, str.stringval, strlen(str.stringval));
-		cur_len += strlen(str.stringval);
-		memcpy(res + cur_len, delimiter, delimeter_len);
-		cur_len += delimeter_len;
+		int val_len = strlen(str.stringval);
+
+		// Protect from buffer overflow.
+		// Might happen if str_len overflowed
+		cur_len += val_len + delimeter_len;
```

Comment:
I'm not sure this solution is right,
If both `str_len` and `cur_len` had overflowed, the following condition would pass and we will be overflowing the `res` buffer.

Reply:
@swilly22 please recheck